core/scripts/chaincli/handler/keeper_upkeep_events.go

```diff
@@ -0,0 +1,64 @@
+package handler
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/upkeep_counter_wrapper"
+)
+
+//UpkeepCounterEvents print out emitted events and write to csv file
+func (k *Keeper) UpkeepCounterEvents(ctx context.Context, hexAddr string, fromBlock, toBlock uint64) {
+	contractAddress := common.HexToAddress(hexAddr)
+	upkeepCounter, err := upkeep_counter_wrapper.NewUpkeepCounter(contractAddress, k.client)
+	if err != nil {
+		log.Fatal(err)
+	}
+	filterOpts := bind.FilterOpts{
+		Start:   fromBlock,
+		End:     &toBlock,
+		Context: ctx,
+	}
+	upkeepIterator, err := upkeepCounter.FilterPerformingUpkeep(&filterOpts, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	filename := fmt.Sprintf("%s.csv", hexAddr)
+	file, err := os.Create(filename)
+	defer func(file *os.File) { _ = file.Close() }(file)
+	if err != nil {
+		log.Fatalln("failed to open file", err)
+	}
+	w := csv.NewWriter(file)
+	defer w.Flush()
+
+	fmt.Println("From, InitialBlock, LastBlock, PreviousBlock, Counter")
+	row := []string{"From", "InitialBlock", "LastBlock", "PreviousBlock", "Counter"}
```

Comment:
It can be moved to a package level, what do you think?

Reply:
you mean put the static array as a package var? its specific to this even so unless its needed in the future i think its fine here for now